src/org/opendatakit/briefcase/ui/SettingsPanel.java

```diff
@@ -88,17 +89,26 @@ public void actionPerformed(ActionEvent arg0) {
     chkParallel.addActionListener(new ParallelPullToggleListener());
 
     chkTrackingConsent.setSelected(BriefcasePreferences.getBriefcaseTrackingConsentProperty());
-    chkTrackingConsent.addActionListener(new TrackingConsentToggleListener(parentWindow.briefcaseAnalytics));
+    chkTrackingConsent.addActionListener(__ -> {
+      if (chkTrackingConsent.isSelected())
+        analytics.enableTracking();
+      else
+        analytics.disableTracking();
+      BriefcasePreferences.setBriefcaseTrackingConsentProperty(chkTrackingConsent.isSelected());
+
+    });
 
     chkStorePasswordsConsent.setSelected(BriefcasePreferences.getStorePasswordsConsentProperty());
     chkStorePasswordsConsent.addActionListener(e -> {
-      if (e.getSource() == chkStorePasswordsConsent) {
-        boolean isSelected = chkStorePasswordsConsent.isSelected();
-        BriefcasePreferences.setStorePasswordsConsentProperty(isSelected);
-        if (!isSelected) {
-          PullTransferPanel.PREFERENCES.remove(BriefcasePreferences.PASSWORD);
-          PushTransferPanel.PREFERENCES.remove(BriefcasePreferences.PASSWORD);
-        }
+      boolean isSelected = chkStorePasswordsConsent.isSelected();
+      if (isSelected)
+        analytics.event("Settings", "Save passwords", "Enable", null);
+      else
+        analytics.event("Settings", "Save passwords", "Disable", null);
+      BriefcasePreferences.setStorePasswordsConsentProperty(isSelected);
+      if (!isSelected) {
+        PullTransferPanel.PREFERENCES.remove(BriefcasePreferences.PASSWORD);
```

Comment:
Not sure I can sell this, and I bet you already considered it:
```
Arrays.asList(PullTransferPanel.PREFERENCES, PushTransferPanel.PREFERENCES).forEach(p -> 
    p.remove(BriefcasePreferences.PASSWORD));
```
Maybe if we already had a collection of the two panels, shared with other code.

Reply:
Don't worry about this lines. They look much better (and without breaking encapsulation) on #353 (see [here](https://github.com/opendatakit/briefcase/pull/353/files#diff-71c30114fc8ce489cd3b85b0a27e9872R98))